app/controllers/services_controller.rb

```diff
@@ -75,6 +75,28 @@ def deactivate
     end
   end
 
+  def set_right
+    return head(:forbidden) unless current_user.has_role?(:superadmin)
+    service_owner = Service.find(params[:id])
+    service_targeted = Service.find(params[:target_id])
+    service_owner.add_role(:all, service_targeted)
+    redirect_to admin_board_service_path(params[:id])
+  end
+
+  def unset_right
+    return head(:forbidden) unless current_user.has_role?(:superadmin)
+    service_owner = Service.find(params[:id])
+    service_targeted = Service.find(params[:target_id])
+    service_owner.remove_role :all, service_targeted
+    redirect_to admin_board_service_path(params[:id])
+  end
+
+  def admin_board
+    return head(:forbidden) unless current_user.has_role?(:superadmin)
```

Comment:
@bosulliv42 
This line appears three times in that controller, you can refactor your code with a `before_action` callback